renderdoc/driver/vulkan/vk_acceleration_structure.h

```diff
@@ -28,9 +28,52 @@
 
 class WrappedVulkan;
 
-// Just holds the built flag, will eventually hold all the AS build data
 struct VkAccelerationStructureInfo
 {
+  struct GeometryData
+  {
+    struct Triangles
+    {
+      VkFormat vertexFormat;
+      VkDeviceSize vertexStride;
+      uint32_t maxVertex;
+      VkIndexType indexType;
+      bool hasTransformData;
+    };
+
+    struct Aabbs
+    {
+      VkDeviceSize stride;
+    };
+
+    struct Instances
+    {
+      // arrayOfPointers TODO
+    };
```

Comment:
Ah sorry I misunderstood what this is for.

With `arrayOfPointers` what we want to store isn't different we just can't do it in one copy. We'll probably need to do an inefficient compute-based copy to follow the indirections, but in that case this struct still wouldn't be needed as the data would still end up in `readbackMem` just via a more circuitous route?

Reply:
Ah yes, you're absolutely right as the copy would 'flatten' the indirection so on replay `arrayOfPointers` would always be false - I'll remove it.

Reply:
I'm going to need some hand holding when it comes to using a compute shader for the indirect copying - is there anything similar in the codebase already I can base it on?

Reply:
I don't think there's anything similar on vulkan unfortunately. This is something uniquely difficult on RT needing this kind of GPU work at capture time.

Roughly it would mean tracking enough compute state in the command buffer during capture to push + pop around our own dispatch, and then write a shader to indirect through the pointers and copy out a BLAS-instance sized bit of memory to the output. I think since you will inevitably disturb the pipeline layout and push constants etc you can push-constant the number of BLASs.

Reply:
For this patch would it be acceptable to just fail the capture if `arrayOfPointers` is set and then return to the issue after the general case is merged?  I'm worried about getting bogged down in this when I haven't even seen the mode used yet.

Reply:
Oh yes sorry I didn't mean to imply you needed to add support here. As long as there's an assert that the case isn't handled it can be left as a TODO for a future PR. I only made the original comment because I misunderstood what you were needing this struct for.